packages/task/src/browser/task-service.ts

```diff
@@ -402,10 +432,146 @@ export class TaskService implements TaskConfigurationClient {
             }
         }
 
+        const tasks = await this.getWorkspaceTasks(task._scope);
         const resolvedMatchers = await this.resolveProblemMatchers(task, customizationObject);
-        return this.runTask(task, {
-            customization: { ...customizationObject, ...{ problemMatcher: resolvedMatchers } }
+        try {
+            const rootNode = new TaskNode(task, [], []);
+            this.detectDirectedAcyclicGraph(task, rootNode, tasks);
+            return await this.runTasksGraph(task, tasks, {
+                customization: { ...customizationObject, ...{ problemMatcher: resolvedMatchers } }
+            });
+        } catch (error) {
+            this.logger.error(error.message);
+            this.messageService.error(error.message);
+            return undefined;
+        }
+    }
+
+    async runTasksGraph(task: TaskConfiguration, tasks: TaskConfiguration[], option?: RunTaskOption): Promise<TaskInfo | undefined> {
```

Comment:
Can we add documentation for the following method?
The method is quite large and complex and overall it'd be good to document our intentions and what the code attempts to achieve.

Reply:
fixed.done